cli/command/container/opts.go

```diff
@@ -788,6 +793,12 @@ func applyContainerOptions(n *opts.NetworkAttachmentOpts, copts *containerOption
 	if n.IPv6Address != "" && copts.ipv6Address != "" {
 		return errdefs.InvalidParameter(errors.New("conflicting options: cannot specify both --ip6 and per-network IPv6 address"))
 	}
+	if n.MacAddress != "" && copts.macAddress != "" {
+		return errdefs.InvalidParameter(errors.New("conflicting options: cannot specify both --mac-address and per-network MAC address"))
+	}
```

Comment:
Haven't checked out the code locally yet, but I wonder now if these errors are always correct; this error is _legit_ if I would do the following;


This is a conflict, because I'm setting the `mac-address` _TWICE_ for the default (`bridge`) network;

```bash
docker run \
    --mac-address=foo \
    --network name=bridge,mac-address=bar \
   myimage
```


This is _NOT_ a conflict, because I'm setting the `mac-address` twice, but for different networks; for `bridge`, I set it to `foo` and for `mynetwork`, I set it to `bar`;

```bash
docker run \
    --mac-address=foo \
    --network bridge \
    --network name=mynetwork,mac-address=bar \
   myimage
```


This one is more tricky though, as this one could be seen in 2 different ways;

- an implicit `--network=bridge` with a mac-address specified for the default (`bridge`) network
- a conflict (because I specified the network to be `mynetwork`, NOT `bridge`, and specified the mac-address _twice_ for that network);

```bash
docker run \
    --mac-address=foo \
    --network name=mynetwork,mac-address=bar \
   myimage
```

(Taking `mac-address` as example, but the same applies to other options that also have a separate `--flag`)

Reply:
I think the original idea was to make sure no potentially ambiguous notations could be used. I think it will avoid some mistakes, so it's probably better to stick with that UX.

As it's mostly unrelated to my PR, could you open an issue if you think it's worth revisiting?